From 522cc3a0a06a3c1054f345801b0e9ed3ea5386e3 Mon Sep 17 00:00:00 2001
From: Rich Megginson <rmeggins@redhat.com>
Date: Fri, 23 Jul 2010 16:03:51 -0600
Subject: [PATCH] Bug 617629  - Missing aliases in new schema files

https://bugzilla.redhat.com/show_bug.cgi?id=617629
Resolves: bug 617629
Bug Description: Missing aliases in new schema files
Reviewed by: ???
Branch: master
Fix Description: add back the aliases we removed - added a new schema flag
X-DEPRECATED - this is a quoted string containing a space separated list
of the deprecated aliases for this attribute
Platforms tested: RHEL5 x86_64
Flag Day: no
Doc impact: no
---
 ldap/schema/00core.ldif          |   60 ++++++++++++++++++++++---------------
 ldap/schema/05rfc4524.ldif       |   30 +++++++++++-------
 ldap/schema/06inetorgperson.ldif |    5 ++-
 3 files changed, 57 insertions(+), 38 deletions(-)

diff --git a/ldap/schema/00core.ldif b/ldap/schema/00core.ldif
index 3a02192..7f5dab6 100644
--- a/ldap/schema/00core.ldif
+++ b/ldap/schema/00core.ldif
@@ -317,26 +317,29 @@ attributeTypes: ( 2.5.4.15 NAME 'businessCategory'
 #
 ################################################################################
 #
-attributeTypes: ( 2.5.4.6 NAME 'c'
+attributeTypes: ( 2.5.4.6 NAME ( 'c' 'countryName' )
   SUP name
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.11
   SINGLE-VALUE
-  X-ORIGIN 'RFC 4519' )
+  X-ORIGIN 'RFC 4519'
+  X-DEPRECATED 'countryName' )
 #
 ################################################################################
 #
-attributeTypes: ( 2.5.4.3 NAME 'cn'
+attributeTypes: ( 2.5.4.3 NAME ( 'cn' 'commonName' )
   SUP name
-  X-ORIGIN 'RFC 4519' )
+  X-ORIGIN 'RFC 4519'
+  X-DEPRECATED 'commonName' )
 #
 ################################################################################
 #
-attributeTypes: ( 0.9.2342.19200300.100.1.25 NAME 'dc'
+attributeTypes: ( 0.9.2342.19200300.100.1.25 NAME ( 'dc' 'domaincomponent' )
   EQUALITY caseIgnoreIA5Match
   SUBSTR caseIgnoreIA5SubstringsMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.26
   SINGLE-VALUE
-  X-ORIGIN 'RFC 4519' )
+  X-ORIGIN 'RFC 4519'
+  X-DEPRECATED 'domaincomponent' )
 #
 ################################################################################
 #
@@ -356,10 +359,11 @@ attributeTypes: ( 2.5.4.27 NAME 'destinationIndicator'
 #
 ################################################################################
 #
-attributeTypes: ( 2.5.4.49 NAME 'distinguishedName'
+attributeTypes: ( 2.5.4.49 NAME ( 'distinguishedName' 'dn' )
   EQUALITY distinguishedNameMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.12
-  X-ORIGIN 'RFC 4519' )
+  X-ORIGIN 'RFC 4519'
+  X-DEPRECATED 'dn' )
 #
 ################################################################################
 #
@@ -378,9 +382,10 @@ attributeTypes: ( 2.5.4.47 NAME 'enhancedSearchGuide'
 #
 ################################################################################
 #
-attributeTypes: ( 2.5.4.23 NAME 'facsimileTelephoneNumber'
+attributeTypes: ( 2.5.4.23 NAME ( 'facsimileTelephoneNumber' 'fax' )
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.22
-  X-ORIGIN 'RFC 4519' )
+  X-ORIGIN 'RFC 4519'
+  X-DEPRECATED 'fax' )
 #
 ################################################################################
 #
@@ -418,9 +423,10 @@ attributeTypes: ( 2.5.4.25 NAME 'internationalISDNNumber'
 #
 ################################################################################
 #
-attributeTypes: ( 2.5.4.7 NAME 'l'
+attributeTypes: ( 2.5.4.7 NAME ( 'l' 'locality' 'localityname' )
   SUP name
-  X-ORIGIN 'RFC 4519' )
+  X-ORIGIN 'RFC 4519'
+  X-DEPRECATED 'locality localityname' )
 #
 ################################################################################
 #
@@ -430,15 +436,17 @@ attributeTypes: ( 2.5.4.31 NAME 'member'
 #
 ################################################################################
 #
-attributeTypes: ( 2.5.4.10 NAME 'o'
+attributeTypes: ( 2.5.4.10 NAME ( 'o' 'organizationname' )
   SUP name
-  X-ORIGIN 'RFC 4519' )
+  X-ORIGIN 'RFC 4519'
+  X-DEPRECATED 'organizationname' )
 #
 ################################################################################
 #
-attributeTypes: ( 2.5.4.11 NAME 'ou'
+attributeTypes: ( 2.5.4.11 NAME ( 'ou' 'organizationalUnitName' )
   SUP name
-  X-ORIGIN 'RFC 4519' )
+  X-ORIGIN 'RFC 4519'
+  X-DEPRECATED 'organizationalUnitName' )
 #
 ################################################################################
 #
@@ -520,23 +528,26 @@ attributeTypes: ( 2.5.4.5 NAME 'serialNumber'
 #
 ################################################################################
 #
-attributeTypes: ( 2.5.4.4 NAME 'sn'
+attributeTypes: ( 2.5.4.4 NAME ( 'sn' 'surName' )
   SUP name
-  X-ORIGIN 'RFC 4519' )
+  X-ORIGIN 'RFC 4519'
+  X-DEPRECATED 'surName' )
 #
 ################################################################################
 #
-attributeTypes: ( 2.5.4.8 NAME 'st'
+attributeTypes: ( 2.5.4.8 NAME ( 'st' 'stateOrProvinceName' )
   SUP name
-  X-ORIGIN 'RFC 4519' )
+  X-ORIGIN 'RFC 4519'
+  X-DEPRECATED 'stateOrProvinceName' )
 #
 ################################################################################
 #
-attributeTypes: ( 2.5.4.9 NAME 'street'
+attributeTypes: ( 2.5.4.9 NAME ( 'street' 'streetaddress' )
   EQUALITY caseIgnoreMatch
   SUBSTR caseIgnoreSubstringsMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.15
-  X-ORIGIN 'RFC 4519' )
+  X-ORIGIN 'RFC 4519'
+  X-DEPRECATED 'streetaddress' )
 #
 ################################################################################
 #
@@ -566,11 +577,12 @@ attributeTypes: ( 2.5.4.12 NAME 'title'
 #
 ################################################################################
 #
-attributeTypes: ( 0.9.2342.19200300.100.1.1 NAME 'uid'
+attributeTypes: ( 0.9.2342.19200300.100.1.1 NAME ( 'uid' 'userid' )
   EQUALITY caseIgnoreMatch
   SUBSTR caseIgnoreSubstringsMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.15
-  X-ORIGIN 'RFC 4519' )
+  X-ORIGIN 'RFC 4519'
+  X-DEPRECATED 'userid' )
 #
 ################################################################################
 #
diff --git a/ldap/schema/05rfc4524.ldif b/ldap/schema/05rfc4524.ldif
index e666368..ca6073a 100644
--- a/ldap/schema/05rfc4524.ldif
+++ b/ldap/schema/05rfc4524.ldif
@@ -28,11 +28,12 @@ attributeTypes: ( 0.9.2342.19200300.100.1.48 NAME 'buildingName'
 #
 ################################################################################
 #
-attributeTypes: ( 0.9.2342.19200300.100.1.43 NAME 'co'
+attributeTypes: ( 0.9.2342.19200300.100.1.43 NAME ( 'co' 'friendlycountryname' )
   EQUALITY caseIgnoreMatch
   SUBSTR caseIgnoreSubstringsMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.15
-  X-ORIGIN 'RFC 4524' )
+  X-ORIGIN 'RFC 4524'
+  X-DEPRECATED 'friendlycountryname' )
 #
 ################################################################################
 #
@@ -83,19 +84,21 @@ attributeTypes: ( 0.9.2342.19200300.100.1.13 NAME 'documentVersion'
 #
 ################################################################################
 #
-attributeTypes: ( 0.9.2342.19200300.100.1.5 NAME 'drink'
+attributeTypes: ( 0.9.2342.19200300.100.1.5 NAME ( 'drink' 'favouriteDrink' )
   EQUALITY caseIgnoreMatch
   SUBSTR caseIgnoreSubstringsMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.15{256}
-  X-ORIGIN 'RFC 4524' )
+  X-ORIGIN 'RFC 4524'
+  X-DEPRECATED 'favouriteDrink' )
 #
 ################################################################################
 #
-attributeTypes: ( 0.9.2342.19200300.100.1.20 NAME 'homePhone'
+attributeTypes: ( 0.9.2342.19200300.100.1.20 NAME ( 'homePhone' 'homeTelephoneNumber' )
   EQUALITY telephoneNumberMatch
   SUBSTR telephoneNumberSubstringsMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.50
-  X-ORIGIN 'RFC 4524' )
+  X-ORIGIN 'RFC 4524'
+  X-DEPRECATED 'homeTelephoneNumber' )
 #
 ################################################################################
 #
@@ -123,11 +126,12 @@ attributeTypes: ( 0.9.2342.19200300.100.1.4 NAME 'info'
 #
 ################################################################################
 #
-attributeTypes: ( 0.9.2342.19200300.100.1.3 NAME 'mail'
+attributeTypes: ( 0.9.2342.19200300.100.1.3 NAME ( 'mail' 'rfc822mailbox' )
   EQUALITY caseIgnoreIA5Match
   SUBSTR caseIgnoreIA5SubstringsMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.26{256}
-  X-ORIGIN 'RFC 4524' )
+  X-ORIGIN 'RFC 4524'
+  X-DEPRECATED 'rfc822mailbox' )
 #
 ################################################################################
 #
@@ -138,11 +142,12 @@ attributeTypes: ( 0.9.2342.19200300.100.1.10 NAME 'manager'
 #
 ################################################################################
 #
-attributeTypes: ( 0.9.2342.19200300.100.1.41 NAME 'mobile'
+attributeTypes: ( 0.9.2342.19200300.100.1.41 NAME ( 'mobile' 'mobileTelephoneNumber' )
   EQUALITY telephoneNumberMatch
   SUBSTR telephoneNumberSubstringsMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.50
-  X-ORIGIN 'RFC 4524' )
+  X-ORIGIN 'RFC 4524'
+  X-DEPRECATED 'mobileTelephoneNumber' )
 #
 ################################################################################
 #
@@ -154,11 +159,12 @@ attributeTypes: ( 0.9.2342.19200300.100.1.45 NAME 'organizationalStatus'
 #
 ################################################################################
 #
-attributeTypes: ( 0.9.2342.19200300.100.1.42 NAME 'pager'
+attributeTypes: ( 0.9.2342.19200300.100.1.42 NAME ( 'pager' 'pagerTelephoneNumber' )
   EQUALITY telephoneNumberMatch
   SUBSTR telephoneNumberSubstringsMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.50
-  X-ORIGIN 'RFC 4524' )
+  X-ORIGIN 'RFC 4524'
+  X-DEPRECATED 'pagerTelephoneNumber' )
 #
 ################################################################################
 #
diff --git a/ldap/schema/06inetorgperson.ldif b/ldap/schema/06inetorgperson.ldif
index 9eb9765..96474f2 100644
--- a/ldap/schema/06inetorgperson.ldif
+++ b/ldap/schema/06inetorgperson.ldif
@@ -109,11 +109,12 @@ attributeTypes: ( 0.9.2342.19200300.100.1.7 NAME 'photo'
 #
 ################################################################################
 #
-attributeTypes: ( 1.3.6.1.4.1.250.1.57 NAME 'labeledURI'
+attributeTypes: ( 1.3.6.1.4.1.250.1.57 NAME ( 'labeledURI' 'labeledurl' )
   EQUALITY caseExactMatch
   SUBSTR caseExactSubstringsMatch
   SYNTAX 1.3.6.1.4.1.1466.115.121.1.15
-  X-ORIGIN 'RFC 2079' )
+  X-ORIGIN 'RFC 2079'
+  X-DEPRECATED 'labeledurl' )
 #
 ################################################################################
 #
-- 
1.5.5.6

